iwyu_ast_util.h

```diff
@@ -79,6 +81,16 @@ using std::string;
 // we store whether a node is in a 'forward-declarable' context
 // (such as a function parameter), meaning all types seen below
 // that node are legal to fowrard-declare according to c++.
+
+namespace detail {
+template <typename T>
+const T* null_value = nullptr;
+
+template <>
+inline clang::NestedNameSpecifier null_value<clang::NestedNameSpecifier> =
+    std::nullopt;
```

Comment:
Braced initialization would be more robust against possible refactoring if the NNS constructor suddenly becomes explicit. But maybe, copy-initialization looks more readable.